br/pkg/task/restore.go

```diff
@@ -1166,6 +1176,134 @@ func runRestore(c context.Context, g glue.Glue, cmdName string, cfg *RestoreConf
 	return nil
 }
 
+func getMaxReplica(ctx context.Context, mgr *conn.Mgr) (cnt uint64, err error) {
+	var resp map[string]any
+	err = utils.WithRetry(ctx, func() error {
+		resp, err = mgr.GetPDHTTPClient().GetReplicateConfig(ctx)
+		return err
+	}, utils.NewPDReqBackoffer())
+	if err != nil {
+		return 0, errors.Trace(err)
+	}
+
+	key := "max-replicas"
+	val, ok := resp[key]
+	if !ok {
+		return 0, errors.Errorf("key %s not found in response %v", key, resp)
+	}
+	return uint64(val.(float64)), nil
+}
+
+func getStores(ctx context.Context, mgr *conn.Mgr) (stores *http.StoresInfo, err error) {
+	err = utils.WithRetry(ctx, func() error {
+		stores, err = mgr.GetPDHTTPClient().GetStores(ctx)
+		return err
+	}, utils.NewPDReqBackoffer())
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	return stores, nil
+}
+
+func EstimateTikvUsage(files []*backuppb.File, maxReplica uint64, storeCnt int) uint64 {
+	if storeCnt == 0 {
+		return 0
+	}
+	var totalSize uint64 = 0
+	for _, file := range files {
+		totalSize += file.GetSize_()
+	}
+	return totalSize * maxReplica / uint64(storeCnt)
+}
+
+func EstimateTiflashUsage(tables []*metautil.Table, storeCnt int) uint64 {
+	if storeCnt == 0 {
+		return 0
+	}
+	var tiflashTotal uint64 = 0
+	for _, table := range tables {
+		if table.TiFlashReplicas <= 0 {
```

Comment:
table.TiFlashReplicas seems always be 0, since this field not set in backup.

Reply:
resolved